grakn-core/src/main/java/ai/grakn/util/Schema.java

```diff
@@ -145,11 +151,11 @@ public Class getClassType(){
      */
     public enum ConceptProperty {
         //Unique Properties
-        TYPE_LABEL(String.class), INDEX(String.class), ID(String.class),
+        TYPE_LABEL(String.class), INDEX(String.class), ID(String.class), TYPE_ID(Integer.class),
 
         //Other Properties
-        TYPE(String.class), IS_ABSTRACT(Boolean.class), IS_IMPLICIT(Boolean.class), IS_SHARD(Boolean.class),
-        REGEX(String.class), DATA_TYPE(String.class), INSTANCE_COUNT(Long.class),
+        INSTANCE_TYPE_ID(Integer.class), IS_ABSTRACT(Boolean.class), IS_IMPLICIT(Boolean.class), IS_SHARD(Boolean.class),
+        REGEX(String.class), DATA_TYPE(String.class), INSTANCE_COUNT(Long.class), CURRENT_TYPE_ID(Integer.class),
```

Comment:
`CURRENT_TYPE_ID ` ? You mean `TYPE_ID`? 

`TYPE_ID` - Is only stored in types and contains the integer id we assign to types. It is indexed to be unique.

`INSTANCE_TYPE_ID ` - Is only stored on instances and refers to the `TYPE_ID` of the instance's type. This is NOT indexed in any way. Hence why we have two properties.